autorest/models/name_converter.py

```diff
@@ -0,0 +1,186 @@
+            for schema in schema_yamls:
+                if self._seen_id(id(schema)):
+                    continue
+                if type_list == 'objects':
+                    continue
+                if type_list == 'arrays' or type_list == 'arrays':
+                    NameConverter._convert_language_default_name(schema)
+                    NameConverter._convert_language_default_name(schema['elementType'])
+                elif type_list == 'constants':
+                    NameConverter._convert_language_default_name(schema)
+                    NameConverter._convert_language_default_name(schema['value'])
+                    NameConverter._convert_language_default_name(schema['valueType'])
+                elif type_list == 'sealedChoices' or type_list == 'choices':
+                    self._convert_enum_schema(schema)
+                else:
+                    NameConverter._convert_language_default_name(schema)
+
+    def _convert_enum_schema(self, schema):
+        schema['language']['python'] = dict(schema['language']['default'])
+        schema_name = schema['language']['default']['name']
+        schema_python_name = schema['language']['python']['name']
+        schema_python_name = NameConverter._to_camel_case(schema_name)
+        schema['language']['python']['name'] = schema_python_name
+
+        NameConverter._convert_language_default_name(schema['choiceType'])
+
+        for choice in schema['choices']:
+            if self._seen_id(id(choice)):
+                continue
+            NameConverter._convert_language_default_name(choice, pad_string='Enum')
+
+    def _convert_object_schema(self, schema):
+        schema['language']['python'] = dict(schema['language']['default'])
+        schema_name = schema['language']['default']['name']
+        schema_python_name = schema['language']['python']['name']
+
+        schema_python_name = NameConverter._to_camel_case(schema_name)
+        schema['language']['python']['name'] = schema_python_name
+        for prop in schema.get('properties', []):
+            if self._seen_id(id(prop)):
+                continue
+            NameConverter._convert_language_default_name(schema=prop, pad_string='Property')
+
+    @staticmethod
+    def _convert_language_default_name(schema, **kwargs):
+        schema['language']['python'] = dict(schema['language']['default'])
+        schema_name = schema['language']['default']['name']
+        schema_python_name = schema['language']['python']['name']
+
+        schema_python_name = NameConverter._to_valid_python_name(schema_name, **kwargs)
+        schema['language']['python']['name'] = schema_python_name
+
+
+    @staticmethod
+    def _to_camel_case(name):
+        name_list = re.split('[^a-zA-Z\\d]', name)
+        name_list = [s[0].upper() + s[1:] if len(s) > 1 else s.upper()
+                            for s in name_list]
+        return ''.join(name_list)
+
+    @staticmethod
+    def _to_valid_python_name(name, **kwargs):
```

Comment:
Why kwargs? This is Python 3 code, please use the "*" syntax

Reply:
Also, Please add unittests with all simple and corner case scenario

Reply:
added